crates/ruff_linter/src/rules/ruff/rules/float_literal_decimal.rs

```diff
@@ -0,0 +1,76 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast};
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `Decimal` calls passing a float literal.
+///
+/// ## Why is this bad?
+/// Float literals are non-deterministic and can lead to unexpected results. The `Decimal` class is designed to handle
+/// numbers with fixed-point precision, so a string literal should be used instead.
+///
+/// ## Example
+///
+/// ```python
+/// num = Decimal(1.2345)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// num = Decimal("1.2345")
+/// ```
+#[violation]
+pub struct FloatLiteralDecimal;
```

Comment:
I find it difficult to understand what the rule `FloatLiteralDecimal` lints against because it's just a sequence of nouns.

How about `DecimalWithFloatValue`?

A more involved implementation could resolve the floats runtime value and compare if it is the same as the literal. This might even be a useful rule outside the context of `Decimal`. @AlexWaygood wdyt?

Reply:
> I find it difficult to understand what the rule `FloatLiteralDecimal` lints against because it's just a sequence of nouns.
> 
> How about `DecimalWithFloatValue`?

I'd vote for `DecimalFromFloatLiteral` (since the `Decimal` object being constructed won't ultimately wrap a "float value" at all).

> A more involved implementation could resolve the floats runtime value and compare if it is the same as the literal. This might even be a useful rule outside the context of `Decimal`. @AlexWaygood wdyt?

For now I think I'd prefer to keep this a simple rule specific to `decimal.Decimal`, since this is a well-known footgun. We could maybe consider a more involved rule separately.

Reply:
The problem with considering a more involved rule separately is that the rules then have overlapping concerns which is something we generally want to avoid. That's why we may have to decide this now unless we're fine with redirecting later.

Reply:
I'm still inclined to go for a specific rule that fixes a known problem where we can have a high degree of accuracy. It's unclear to me exactly how the more general rule would work and I doubt we'd have the same degree of accuracy trying to generalise it.